src/agent_workflow_server/services/runs.py

```diff
@@ -247,6 +253,37 @@ async def wait_for_output(run_id: str, timeout: float = None):
 
         return None, None
 
+    @staticmethod
+    async def stream_events(run_id: str) -> AsyncIterator[StreamEventPayload | None]:
+        async for message in Runs.Stream.join(run_id):
+            msg_data = message.data
+
+            if message.type == "control":
+                if message.data == "done":
+                    break
+                elif message.data == "timeout":
+                    yield None
+                    continue
+                else:
+                    logger.error(
+                        f'received control message "{message.data}" in stream events for run: {run_id}'
```

Comment:
```suggestion
                        f'received unknown control message "{message.data}" in stream events for run: {run_id}'
```
nit ^ improve error message